x/emissions/keeper/inference_synthesis/network_inference_builder.go

```diff
@@ -7,136 +7,99 @@ import (
 	"cosmossdk.io/log"
 	alloraMath "github.com/allora-network/allora-chain/math"
 	emissions "github.com/allora-network/allora-chain/x/emissions/types"
-	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
-type NetworkInferenceBuilder struct {
-	ctx     sdk.Context
-	logger  log.Logger
-	palette SynthPalette
-	// Network Inferences Properties
-	inferences                 []*emissions.WorkerAttributedValue
-	forecastImpliedInferences  []*emissions.WorkerAttributedValue
-	weights                    RegretInformedWeights
-	combinedInference          InferenceValue
-	naiveInference             InferenceValue
-	oneOutInfererInferences    []*emissions.WithheldWorkerAttributedValue
-	oneOutForecasterInferences []*emissions.WithheldWorkerAttributedValue
-	oneInInferences            []*emissions.WorkerAttributedValue
-}
-
-func NewNetworkInferenceBuilderFromSynthRequest(
-	req SynthRequest,
-) (*NetworkInferenceBuilder, error) {
-	paletteFactory := SynthPaletteFactory{}
-	palette, err := paletteFactory.BuildPaletteFromRequest(req)
-	if err != nil {
-		return nil, errorsmod.Wrapf(err, "Error building palette from request")
-	}
-	return &NetworkInferenceBuilder{
-		ctx:     req.Ctx,
-		logger:  Logger(req.Ctx),
-		palette: palette,
-	}, nil
-}
-
 // Calculates the network combined inference I_i, Equation 9
-func (b *NetworkInferenceBuilder) SetCombinedValue() *NetworkInferenceBuilder {
-	b.logger.Debug(fmt.Sprintf("Calculating combined inference for topic %v", b.palette.TopicId))
-	palette := b.palette.Clone()
+func GetCombinedInference(logger log.Logger, inPalette SynthPalette) (
+	weights RegretInformedWeights, combinedInference InferenceValue, err error) {
+	logger.Debug(fmt.Sprintf("Calculating combined inference for topic %v", inPalette.TopicId))
+	palette := inPalette.Clone()
```

Comment:
I think we don't need to clone the palettes in this case and in other cases below, please double check!

Reply:
Looked into this, it seems the palette cloning is necessary - otherwise the CSV simulator values don't match - so something about the way the data is shared means the copy is necessary. I am looking more into it